### PR TITLE
fix testcontainers

### PR DESCRIPTION
**Pull request by jianyun8023**
**The testcontainers version 1.14.3 used failed to test because of a dependent docker image**. 
testcontainers issue: ["No such image: testcontainers/ryuk:0.3.0"](https://github.com/testcontainers/testcontainers-java/issues/3574)
The solution is to upgrade to 1.15.2.

Since pulsar-flink is using testcontainers indirectly through streamnative-test, it is also necessary to upgrade streamnative-test version.

During this process streamnative-test needs to be released, and pulsar-flink needs to follow suit with a number of changes. After waiting for the streamnative-test release and feature verification, there were still problems. The unit test fix was seriously delayed.

The purpose of pulsar-flink using them is to start a pulsar standalone service for testing. Using only testcontainers is sufficient.

### Solution :
 - Upgrade the testcontainers version to
 - Remove the streamnative-test dependency.
 - Modify test to use testcontainers to start the pulsar standalone service directly in test.

### CODE REVIEW

**Comment by jianyun8023** [comment]
@jiazhai Can you review it for me?

**Comment by jiazhai** [comment]
> fix testcontainers and remove sn-test.

@jianyun8023 Please add more description for this PR, why it failed, why your fix work for this?

File: pulsar-flink-connector/src/test/java/org/apache/flink/streaming/connectors/pulsar/PulsarAuthTest.java [hunk 1]
```diff
@@ -68,29 +64,21 @@ public static void prepare() throws Exception {
...
+                .withClasspathResourceMapping("pulsar/auth-standalone.conf", "/pulsar/conf/standalone.conf",
+                        BindMode.READ_ONLY);
+        pulsarService.withClasspathResourceMapping("pulsar/auth-client.conf", "/pulsar/conf/client.conf",
+                BindMode.READ_ONLY);
+        pulsarService.start();
+        pulsarService.followOutput(new Slf4jLogConsumer(log));
+        serviceUrl = pulsarService.getPulsarBrokerUrl();
+        adminUrl = pulsarService.getHttpServiceUrl();
         Thread.sleep(30 * 1000);
         log.info("-------------------------------------------------------------------------");
```
Comment (codelipenghui) [inline pulsar-flink-connector/src/test/java/org/apache/flink/streaming/connectors/pulsar/PulsarAuthTest.java:80]: Can we avoid the sleep and the log without any meaning?

**Review by codelipenghui — COMMENTED** [review]
Please avoid use sleep, this will introduce an unstable test and make the test run for long time. Here is an example to use the Awaitility to instead the sleep. https://github.com/apache/pulsar/pull/8557/files

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**